lib/web_ui/lib/src/engine/clipboard.dart

```diff
@@ -0,0 +1,105 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+part of engine;
+
+/// Handles clipboard related platform messages.
+class ClipboardMessageHandler {
+  /// Helper to handle copy to clipboard functionality.
+  final CopyToClipboardStrategy _copyToClipboardStrategy;
+
+  /// Helper to handle copy to clipboard functionality.
+  final PasteFromClipboardStrategy _pasteFromClipboardStrategy;
+
+  ClipboardMessageHandler()
+      : this._copyToClipboardStrategy =
+            (html.window.navigator.clipboard.writeText != null)
+                ? ClipboardAPICopyStrategy()
+                : ExecCommandCopyStrategy(),
```

Comment:
Generally, what I do is add a factory constructor in the abstract `CopyToClipboardStrategy` that knows which class to instantiate. 
```dart
abstract class CopyToClipboardStrategy {
  factory CopyToClipboardStrategy() {
    return (html.window.navigator.clipboard.writeText != null)
      ? ClipboardAPICopyStrategy()
      : ExecCommandCopyStrategy();
  }

  /// Places the text onto the browser Clipboard.
  void setData(String text);
}
```

The reason I prefer this is so that callsites don't have to know the details of how to choose the right class to instantiate.

Reply:
done.